pkg/registry/core/node/strategy.go

```diff
@@ -82,6 +83,27 @@ func (nodeStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Objec
 	if !utilfeature.DefaultFeatureGate.Enabled(features.DynamicKubeletConfig) && !nodeConfigSourceInUse(oldNode) {
 		newNode.Spec.ConfigSource = nil
 	}
+	dropDisabledFields(newNode, oldNode)
+}
+
+func dropDisabledFields(node *api.Node, oldNode *api.Node) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) && !multiNodeCIDRsInUse(oldNode) {
```

Comment:
This is subtle, could use a comment.  If I understand it:

// If the oldNode already has multiple PodCIDRs, leave them (even if the gate says otherwise).  If the gate is disabled and the feature is not already in use, only allow 1 PodCIDR.

That said, it's a great corner case.  I need to consult with other reviewers if there is precedent here.

Reply:
Forgot to update: It looks like this is aligned with guidance.  Not requesting changes for now.